### ISSUE TITLE
Possible memory bug in spdlog_ex ?

### ISSUE THREAD

**Issue by gajanak** [gabime/spdlog#797]
I noticed that spdlog_ex, inherit from runtime_error overwrite  `const char* what()` and returns pointer to function local memory_buffer. 

In my usage case,   a catched exception returns me an garbage string.

It could be that this, is a side effect of something else, but if anyone get similar problem... the following diff make the format early, and let runtime_error save the string internal. (It's not easy to save a string without allowed to throw exception ;)

here the diff to common.h spdlog_ex:
```diff
diff --git a/include/spdlog/common.h b/include/spdlog/common.h
index 33e1378..913dada 100644
--- a/include/spdlog/common.h
+++ b/include/spdlog/common.h
@@ -179,24 +179,14 @@ public:
     {
     }
     spdlog_ex(const std::string &msg, int last_errno)
-        : runtime_error(msg)
+        : runtime_error([&](){
+            fmt::memory_buffer buf;
+            fmt::format_system_error(buf, last_errno, msg);
+            return fmt::to_string(buf);
+        }())
         , last_errno_(last_errno)
     {
     }
-    const char *what() const SPDLOG_NOEXCEPT override
-    {
-        if (last_errno_)
-        {
-            fmt::memory_buffer buf;
-            std::string msg(runtime_error::what());
-            fmt::format_system_error(buf, last_errno_, msg);
-            return fmt::to_string(buf).c_str();
-        }
-        else
-        {
-            return runtime_error::what();
-        }
-    }
 
 private:
     int last_errno_{0};
```

**Comment by gabime** [comment]
yes. seems like a  bug.  will be fixed soon

**Comment by gabime** [comment]
Fixed in commit b962fbb15c9f1fb2f5410ff0cf3a4065088c717e

Thanks for reporting.
